### PR TITLE
Fixing some piping issue. https://github.com/Azure/azure-powershell/issues/2996

### PR DESCRIPTION
**Pull request by diwudd**
## Description

The issue was when removing a set of resources, if one of operations failed, the following ones also failed because the "cancellationSource" was set to null by the first failed operation. By reinitializing "cancellationSource", we make the operations "stateless" even in pipeline. 

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270%28v=vs.85%29.aspx).
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).
### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.
### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.
### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@diwudd__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (wud). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by vivsriaus** [comment]
@diwudd can we please add a scenario test for this?

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by shahabhijeet** [comment]
@vivsriaus can you review this change

**Comment by vivsriaus** [comment]
@azuresdkci retest this please

**Comment by diwudd** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ResourceTests.cs [hunk 1]
```diff
@@ -139,5 +139,19 @@ public void TestManageResourceWithZones()
         {
             ResourcesController.NewInstance.RunPsTest("Test-ManageResourceWithZones");
         }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestRemoveAResourceTest()
```
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ResourceTests.cs]: Is the change covered by these two test cases?

**Comment by vivsriaus** [comment]
LGTM

**Comment by cormacpayne** [comment]
@TianoMS Hey Tian, do you sign-off on these changes?

**Comment by TianoMS** [comment]
@cormacpayne Signed off.

**Comment by markcowl** [comment]
on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1255/

LGTM once the on-demand build passes

**Pull request merged**